### PR TITLE
Remove OpenSSL

### LINKED ISSUE #12530: Remove OpenSSL for non-QT builds
OpenSSL is only used for (if I understand correctly):

1. Handling BIP-70 payment requests (can be disabled with `without-bip70`)
2. Generating random numbers (see #10299, #14623)

If (2) can be solved, then non-QT builds no longer need OpenSSL (move `openssl` to `qt_packages` in `depends/packages/packages.mk`).

This potentially makes it easier to build the depends for other platforms such as iOs (see #11720), if they don't need QT.

### PR DESCRIPTION
**Pull request by fanquake**
Now that #17165 has been merged, removing our remaining OpenSSL usage is possible.

That remaining usage was a call to [`RAND_bytes`](https://www.openssl.org/docs/manmaster/man3/RAND_bytes.html) during the ::SLOW path of [ProcRand](https://github.com/bitcoin/bitcoin/blob/master/src/random.cpp#L616). As well as feeding output from our RNG back into OpenSSL via [`RAND_add`](https://www.openssl.org/docs/manmaster/man3/RAND_add.html) during the ::SLOW and ::SLEEP paths.

Optimistically tagged for `0.20.0`. Needs discussion, potentially in an upcoming weekly meeting?

Closes #12530.

### CODE REVIEW

**Comment by jnewbery** [comment]
Concept ACK!

File: src/random.cpp [hunk 1]
```diff
@@ -11,7 +11,7 @@
 #include <compat.h> // for Windows API
 #include <wincrypt.h>
 #endif
-#include <logging.h>  // for LogPrint()
+#include <logging.h>  // for LogPrintf()
 #include <sync.h>     // for WAIT_LOCK
```
Comment (fanquake) [inline src/random.cpp]: note: Will update 5cc4f20c7064fee90a78a827a9e30b2934ce4ca7 to change `WAIT_LOCK` to `Mutex`.

**Comment by sipa** [comment]
I think we wanted to include some of the environmental entropy sources (statistics, pid, ...) that OpenSSL uses in our own RNG state first. See #10299. I'll PR something soon.

**Comment by promag** [comment]
Concept ACK.

**Comment by BlockMechanic** [comment]
Concept ACK.

Irecently ran into openssl issues here https://github.com/bitcoin/bitcoin/issues/17123, this is awesome !

**Comment by laanwj** [comment]
Concept and code review ACK, agree that we should ideally get #17270 in first.

**Comment by practicalswift** [comment]
Concept ACK

Very pleased to see OpenSSL go :)

**Comment by fanquake** [comment]
Fixed doc nit above and squashed some commits together. This is waiting on #17270.

**Comment by Sjors** [comment]
Concept ACK after #17270. This fixes #12530.

**Comment by jamesob** [comment]
big Concept ACK

**Comment by MarcoFalke** [comment]
ACK e5a0bece6e84402fcb1fe4f25fd24da1d21ec077

**Comment by laanwj** [comment]
ACK e5a0bece6e84402fcb1fe4f25fd24da1d21ec077

**Comment by sidhujag** [comment]
concept ACK, is libsecp256k1 going to remove OpenSSL as well? It uses $CRYPTO_CFLAGS

**Comment by sipa** [comment]
@sidhujag libsecp256k1 only uses OpenSSL as an optional dependency in tests. This may change in the future, but the same reasons really don't apply.

**Comment by laanwj** [comment]
it doesn't affect any of the bitcoin core binaries, so it's off topic here. Please take your question upstream.
(That said, cross-checking against other libs is generally a good idea for cryptographic libraries.)

**Review by RdeWilde — COMMENTED** [review]
Great 👌🏼

**Pull request merged**